### PR TITLE
Issue 4 create basic eks cluster

### PR DESCRIPTION
**Pull request by steveellis**
Right now this is deploying nodes to the public subnets rather than the private ones. So still working on that.

### CODE REVIEW

**Comment by steveellis** [comment]
Hitting a bit of a wall on the private/public subnet thing. The problem is that there are 2 pulumi apis for working with eks clusters, [classic](https://www.pulumi.com/registry/packages/aws/api-docs/eks/cluster/) and [something newer](https://www.pulumi.com/registry/packages/eks/api-docs/cluster/) which seems to be related to the newer "aws native" stuff or the "crosswalk" stuff -- really hard to tell. The stuff you're noticing with the properties to set public and private subnet ids, is only available on the newer api. I noticed all these different APIs last week and was a bit distressed by it.

I see a few patterns:
* All of the code sample project  are in "classic", except for 3 that don't help us that are in JS.
* Their code examples in their docs (not the actual API docs) contain snippets from either aws native or their "crosswalk" product.
* Crosswalk is their "better api" in that it does  more stuff for you, but obscures other stuff. They've chosen to fold crosswalk into their "language apis" but the status of that work is unclear.
* There are more code examples in the docs in js.
* "AWS native" is their 1 for 1 match between AWS and their language APIs but not all stuff has been implemented. For example I don't see a`Subnet` class. I noticed a code example that mixes both classic and aws native because of this. 

In any case these are just observations. I think if we want to move forward with "classic" we may need to abandon the private subnets idea with pulumi. I could attempt to re-implement everything in "native". I've tried mixing native and classic without success. Tried passing the private subnet ids to the `NodeGroup` thinking it would deploy to them instead, but that just puts me back to the earlier problem of the nodes not being able to join the cluster.

Basically my takeaway is something like this: dividing a vpc up into public and private subnets was a requirement that the opted to implement in one of their newer apis, but sadly we don't have as good of a code example for that. Could be smooth sailing but I'm already seeing weirdnesses like there not being a `Subnet` class in aws native.

**Comment by steveellis** [comment]
Wanted to link to the [crosswalk stuff and language support](https://www.pulumi.com/docs/guides/crosswalk/aws/#what-languages-are-supported). It really seems like that part of their codebase is where the kinds of benefits that we're looking for from pulumi live.

**Comment by jwfuller** [comment]
Superseded by https://github.com/culibraries/folio/pull/90
